pytorch/pytorch
# [namedtensor] Add test/test_namedtensor.py

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#20168 [namedtensor] Add test/test_namedtensor.py**
* #20163 Add namedtensor build and test to the CI
* #20162 Add USE_NAMEDTENSOR compilation flag.

Tests are skipped if pytorch is compiled without named tensor support
(default).

Test Plan

Compile without namedtensor support
- python test/run_test.py -v --include namedtensor
- Check the output for "skipped because not compiled with named tensor

Compile with namedtensor support
- python test/run_test.py -v --include namedtensor
- Check the output for success on TestNamedTensor.test_trivial

The OSS CI should compile with namedtensor support for this PR.

gh-metadata: pytorch pytorch 20168 gh/zou3519/34/head

Differential Revision: [D15278222](https://our.internmc.facebook.com/intern/diff/D15278222)
